hips/hip-9999.md

```diff
@@ -0,0 +1,254 @@
+---
+hip: 9999
+title: "New annotations for displaying hook output"
+authors: [ "Ian Zink <ian@replicated.com>" ]
+created: "2023-01-26"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This proposes a new annotation to indicate that the output from the hook should be displayed to the user.
+
+## Motivation
+
+The one motivation for this HIP is the ability to display Preflight checks before the main chart installs. This provides vital feedback to the user as to why the chart install is not proceeding.
+
+Often it is important to verify that the kubernetes cluster you are deploying a helm chart into has certain properties. You might need to know that it has an ingress controller available, a certain amount of ephemeral storage, memory, or CPUs available. You might want to validate the the service key they provided was correct or that that database they entered is reachable. Letting a chart deploy and then finding debugging to see why it failed is a poor user experience. These things can all be done with preflight checks enabled by the hook proposed in this HIP.
+
+Another common motivation for this HIP is to indicate that database migration output running as a hook should be visible to the user in the case of failure.
+
+In general, allowing chart developers to run jobs and present that feedback directly to the users could also open up additional use cases beyond just the preflight use case that motivated this HIP.  I could imagine scenarios where maybe CVE warnings are presented or specific upgrade feedback is presented instead of just helm install failure.
```

Comment:
```suggestion
In general, allowing chart developers to run jobs and present that feedback directly to the users could also open up additional use cases beyond just the checks use case that motivated this HIP.  I could imagine scenarios where maybe CVE warnings are presented or specific upgrade feedback is presented instead of just a Helm install failure.
```